### PR TITLE
Fix-dismiss-button-in-full-screen-carousel

### PR DESCRIPTION
**Pull request by scoder1747**
## Summary

The close button becomes invisible at a scenario when a white backgrounded image is presented from Full Screen Carousel view. Here's an example:

IMG_1734

This PR fix the problem by making a dark backgrounded circle around the dismiss button. The reason behind using dark color is because that portion of the view remains dark unless zoom or image-movement is performed.

Screenshot 2024-10-23 at 11 11 26 AM

## Checklist

- [ ] I have read (or I am familiar with) the [Contribution Guidelines](../docs/CONTRIBUTING.md)
- [x] I have tested the changes in this PR
- [ ] My PR is either small, or I have split it into smaller logical commits that are easier to review
- [ ] I have added the signoff line to all my commits. See [Signing off your work](../docs/CONTRIBUTING.md#sign-your-work---the-developers-certificate-of-origin)
- [ ] I have added appropriate changelog entries for the changes in this PR. See [Adding changelog entries](../docs/CONTRIBUTING.md#add-changelog-changed-changelog-fixed-etc)
    - [ ] I do not need to add a changelog entry. Reason: _[Please provide a reason]_
- [ ] I have added appropriate `Closes:` or `Fixes:` tags in the commit messages wherever applicable, or made sure those are not needed. See [Submitting patches](https://github.com/damus-io/damus/blob/master/docs/CONTRIBUTING.md#submitting-patches)

## Other notes

_[Please provide any other information that you think is relevant to this PR.]_

### CODE REVIEW

**Comment by jb55** [comment]
great!

**Comment by jb55** [comment]
can you please format your commits according to our Contributor guidelines?

example:

```
fix: dismiss button in fullscreen carousel

This updates the dismiss button on the fullscreen carousel
so that it is more visible in some scenarios

Changelog-Fixed: Fix dismiss button visibility
Signed-off-by: scoder <scoder@gmail.com>
```

**Comment by scoder1747** [comment]
Hi @jb55, I have amended my commit message as suggested by you

File: damus/Views/Images/ProfilePicImageView.swift [hunk 1]
```diff
@@ -40,29 +40,36 @@ struct ProfileImageContainerView: View {
...
+        self.navDismissBarContainer = navDismissBarContainer
     }
     
     var body: some View {
         HStack {
             Button(action: {
                 presentationMode.wrappedValue.dismiss()
             }, label: {
-                if showBackgroundCircle {
+                if case .profilePicImageView = navDismissBarContainer {
```
Comment (jb55) [inline damus/Views/Images/ProfilePicImageView.swift:62]: if case ... else case is an antipattern if you're covering all cases. use a switch statement instead.

**Review by jb55 — CHANGES_REQUESTED** [review]

File: damus/Views/Images/ProfilePicImageView.swift (hunk 1 above)
Comment (scoder1747) [reply damus/Views/Images/ProfilePicImageView.swift]: @jb55 I have addressed and pushed the change

**Pull request merged**